### PR TITLE
[release-v1.6] Retry Only If the Status Code Is Retryable (#2798)

### PR DESCRIPTION
**Pull request by pierDipi**
Cherry-pick of https://github.com/knative-sandbox/eventing-kafka-broker/commit/3d84363ad294356f15717d6585e1f7b2ed403fec

### CODE REVIEW

**Review by creydr — APPROVED** [review]
/lgtm

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java [hunk 1]
```diff
@@ -133,6 +133,7 @@ private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retr
           if (isRetryableStatusCode(response.statusCode()) && retryCounter < consumerVerticleContext.getEgressConfig().getRetry()) {
             return retry(retryCounter, event);
           }
+          return Future.failedFuture(cause);
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java:136]: @pierDipi did we miss this, when we did some of the refactorings?

Need to port to upstream too?

Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java:136]: It's coming from upstream: https://github.com/knative-sandbox/eventing-kafka-broker/commit/3d84363ad294356f15717d6585e1f7b2ed403fec

**Pull request merged**